src/msg/async/EventKqueue.cc

```diff
@@ -199,13 +199,17 @@ int KqueueDriver::resize_events(int newsize)
   ldout(cct,30) << __func__ << " kqfd = " << kqfd << "newsize = " << newsize 
                 << dendl;
   if(newsize > sav_max) {
-    sav_events = (struct SaveEvent*)realloc( sav_events, 
-                    sizeof(struct SaveEvent)*newsize);
-    if (!sav_events) {
+    void *_realloc = NULL;
+    _realloc = realloc(sav_events,
+                          sizeof(struct SaveEvent)*newsize);
+    if (_realloc == NULL) {
```

Comment:
i am not sure why we need to `free(save_events)` if `realloc()` fails. and this change does not help indeed, the caller (or the caller of its caller) does not handle the return val of it properly. i think maybe the simplest way is to `assert(sav_events)` after `realloc()` call.